DOC-3505: TinyMCE 8.6.0 Documentation (#4134)
tinymce/tinymce-docs: modules/ROOT/pages/self-hosted-trial.adoc

= Self-hosted trial with cloud services
:navtitle: Self-hosted trial
:description: Set up a self-hosted {productname} trial with access to cloud-powered premium features using NPM.
:keywords: trial, self-hosted, npm, cloud services, premium plugins, license key, tinymce-premium

== Overview

A self-hosted trial provides access to {productname} premium plugins in a self-hosted environment. Integrators can optionally connect to {cloudname} services during the trial to evaluate cloud-powered features such as Spell Checker Pro, Export to PDF, and TinyMCE AI, without requiring a separate cloud deployment or additional scripts.

Connecting to cloud services is opt-in. Setting the `+online_license_services_opt_in+` option to `+true+` enables the connection. Without this option, the editor is licensed and operational but cloud-dependent premium plugins are not auto-configured. No separate API key is required.

[NOTE]
====
Only new customers can start a self-hosted trial. For trial duration and entitlement details, see the link:https://www.tiny.cloud/my-account[Tiny Cloud Account] portal.
====

== Prerequisites

> * A {productname} trial account with "self-hosted" selected as the deployment option. Sign up at link:https://www.tiny.cloud/get-tiny/[Get {productname}].

Reviewer: ```suggestion
* A {productname} account with an active trial. Sign up at link:https://www.tiny.cloud/get-tiny/[Get {productname}].
```

Reviewer: Alex flagged that this sounds like there's a choice made when signing up or something, when there isn't

> * A trial license key, issued after sign-up.

Reviewer: ```suggestion
* A trial license key, issued from the 'Self-hosted' tab of the Customer Portal
```

Reviewer: Moving the bit about Self Hosted to this line - seems clearer?

* https://nodejs.org/[Node.js (and npm)] installed.
* A project directory with a `+package.json+` file. Initialize one by running `+npm init+` if needed.
* A secure context: the page must be served over HTTPS or from `+localhost+`. Trial license keys require a secure context to complete key validation.

== Setup

=== Step 1: Install {productname} and premium plugins

Install both the core editor and premium plugins package:

[source,sh,subs="attributes+"]
----
npm install tinymce@^{productmajorversion} tinymce-premium@^{productmajorversion}
----

> === Step 2: Configure the editor with the trial license key
> 
> Add the trial license key to the {productname} configuration using the `+license_key+` option:
> 
> [source,js]
> ----
> tinymce.init({
>   selector: 'textarea',  // change this value according to the HTML
>   license_key: 'your-trial-license-key',
>   plugins: [
>     'advlist', 'autolink', 'lists', 'link', 'image', 'charmap', 'preview',
>     'anchor', 'searchreplace', 'visualblocks', 'code', 'fullscreen',
>     'insertdatetime', 'media', 'table', 'help', 'wordcount',
>     'tinymcespellchecker', 'exportpdf', 'exportword', 'importword',
>     'linkchecker', 'mediaembed', 'editimage', 'autocorrect', 'tinymceai'
>   ],
>   toolbar: 'undo redo | blocks | bold italic | alignleft aligncenter alignright | bullist numlist outdent indent | help',
>   online_license_services_opt_in: true
> });

Reviewer 2: ```suggestion
tinymce.init({
  selector: 'textarea',  // change this value according to the HTML
  license_key: 'your-trial-license-key',
  plugins: [
    'advlist', 'autolink', 'lists', 'link', 'image', 'charmap', 'preview',
    'anchor', 'searchreplace', 'visualblocks', 'code', 'fullscreen',
    'insertdatetime', 'media', 'table', 'help', 'wordcount',
    'tinymcespellchecker', 'exportpdf', 'exportword', 'importword',
    'linkchecker', 'mediaembed', 'editimage', 'autocorrect', 'tinymceai'
  ],
  toolbar: 'undo redo | blocks | bold italic | alignleft aligncenter alignright | bullist numlist outdent indent | help',
  online_license_services_opt_in: true,
  tinymceai_token_provider: async () => {
                await fetch(`https://demo.api.tiny.cloud/1/<your-api-key>/auth/random`, { method: "POST", credentials: "include" });
                return { token: await fetch(`https://demo.api.tiny.cloud/1/<your-api-key>/jwt/tinymceai`, { credentials: "include" }).then(r => r.text()) };
            }
});
```

Reviewer: Noting Alex seconded this suggestion

Reviewer: Seconding this as well, having thought on it more

Author: Added in f693b6428 — `tinymceai_token_provider` is now part of the Step 2 snippet, using the trial demo identity service pattern with a `your-trial-api-key` placeholder. Per the Slack discussion we kept the full config inline so the snippet works end to end, and added a NOTE pointing to the AI JWT page (`#trial-demo-identity-service`) for detail. Thanks for catching the `No token provider configured` error.

----

Replace the `+'your-trial-license-key'+` placeholder with the trial license key provided during sign-up.

> === Step 3: Enable cloud services

Reviewer: Alex flagged that it's a bit weird to have this option set in the snippet and also here as a separate step? Should this just be an explanation/callout of Step 2?

Author: Good call (and Alex) — merged Step 3 into Step 2 in f693b6428. `online_license_services_opt_in` is now only set in the snippet, with the opt-in explanation and the network-connection IMPORTANT callout directly beneath it, so it's no longer configured in two places. Old Step 4 (license key manager) is now Step 3.

= Self-hosted trial with cloud services
:navtitle: Self-hosted trial
:description: Set up a self-hosted {productname} trial with access to cloud-powered premium features using NPM.
:keywords: trial, self-hosted, npm, cloud services, premium plugins, license key, tinymce-premium

== Overview

> A self-hosted trial provides access to {productname} premium plugins in a self-hosted environment. Integrators can optionally connect to {cloudname} services during the trial to evaluate cloud-powered features such as Spell Checker Pro, Export to PDF, and TinyMCE AI, without requiring a separate cloud deployment or additional scripts. Connecting to cloud services is opt-in. Setting the `+online_license_services_opt_in+` option to `+true+` enables the connection. Without this option, the editor is licensed and operational but cloud-dependent premium plugins are not auto-configured.

Reviewer: nitpick, to avoid making it sound like they're only available via cloud:

```suggestion
A self-hosted trial provides access to {productname} premium plugins in a self-hosted environment. Integrators can optionally connect to {cloudname} services during the trial to evaluate server-side features such as Spell Checker Pro, Export to PDF, and TinyMCE AI, without requiring a separate server-side deployment or additional scripts. Connecting to cloud services is opt-in. Setting the `+online_license_services_opt_in+` option to `+true+` enables the connection. Without this option, the editor is licensed and operational but cloud-dependent premium plugins are not auto-configured.
```

[source,js]
----
tinymce.init({
  selector: 'textarea',  // change this value according to the HTML
  license_key: 'your-trial-license-key',  // trial license key from the Self-hosted tab
  plugins: [
    'advlist', 'autolink', 'lists', 'link', 'image', 'charmap', 'preview',
    'anchor', 'searchreplace', 'visualblocks', 'code', 'fullscreen',
    'insertdatetime', 'media', 'table', 'help', 'wordcount',
    'tinymcespellchecker', 'exportpdf', 'exportword', 'importword',
    'linkchecker', 'mediaembed', 'editimage', 'autocorrect', 'tinymceai'
  ],
  toolbar: 'undo redo | blocks | bold italic | alignleft aligncenter alignright | bullist numlist outdent indent | help',
  online_license_services_opt_in: true,
  tinymceai_token_provider: async () => {  // 'no-api-key' is replaced with the Cloud API key once the account is created
    await fetch(`https://demo.api.tiny.cloud/1/no-api-key/auth/random`, { method: 'POST', credentials: 'include' });
    return await fetch(`https://demo.api.tiny.cloud/1/no-api-key/jwt/tinymceai`, { credentials: 'include' })
      .then(resp => resp.text())
      .then(token => ({ token }));
  }
});
----

> Replace the `+'your-trial-license-key'+` placeholder with the trial license key issued from the *Self-hosted* tab of the Customer Portal. The token provider URL uses `+no-api-key+`, a placeholder that is automatically replaced with the Cloud API key once the account is created. The Cloud API key is only visible on the *Cloud* tab of the Customer Portal after the account exists.

Reviewer: similar to the above, I think I'd flip this

```suggestion
Replace the `+'your-trial-license-key'+` placeholder with the trial license key issued from the *Self-hosted* tab of the Customer Portal. The token provider URL uses `+no-api-key+`, a Cloud API key visible on the *Cloud* tab of the Customer Portal. The placeholder is automatically replaced with the Cloud API key by visiting the Customer Portal.
```
or something??

= Self-hosted trial with cloud services
:navtitle: Self-hosted trial
:description: Set up a self-hosted {productname} trial with access to cloud-powered premium features using NPM.
:keywords: trial, self-hosted, npm, cloud services, premium plugins, license key, tinymce-premium

// TODO: Update option name `cloud_services_trial` when final name is confirmed (TINY-14259).

== Overview

A self-hosted trial provides access to {productname} premium plugins and cloud-powered services in a self-hosted environment. This allows evaluation of the full {productname} feature set, including features that depend on cloud services, without requiring a separate cloud deployment or additional scripts.

> During the trial period, {productname} connects to {cloudname} services using the trial license key. No separate API key is required.

Reviewer: ```suggestion
During the trial period, {productname} can be connected to {cloudname} services using the trial license key. No separate API key is required.
```
?

  ],
  toolbar: 'undo redo | blocks | bold italic | alignleft aligncenter alignright | bullist numlist outdent indent | help',
  online_license_services_opt_in: true,

>   tinymceai_token_provider: async () => {  // 'no-api-key' is replaced with the Cloud API key once the account is created

Reviewer: for this comment, I'm torn between this wording and something like "trial API key from the Cloud-hosted tab" to match the one above. mainly thinking of devs who haven't logged in - e.g. got instructions from a manager. or AI.
